core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts

```diff
@@ -897,6 +914,16 @@ export class ConversationSkinComponent {
                     story_url_fragment: storyUrlFragment
                   });
             }
+            this.learnerDashboardBackendApiService
+              .fetchLearnerCompletedChaptersCountDataAsync().then((data) => {
+                let newCompletedChaptersCount = data.completedChaptersCount;
+                if (
+                  newCompletedChaptersCount === this.completedChaptersCount + 1
+                ) {
+                  this.completedChaptersCount = newCompletedChaptersCount;
+                  this.chapterIsCompletedForTheFirstTime = true;
+                }
```

Comment:
I added this check to prevent the message from displaying if the learner replays a chapter. For example if a learner completes 5 chapters, they see the milestone message on completing the 5th chapter. However, if they now replay a completed chapter, they'll again see the message because the total number of chapters is still 5.

With this check, even if the completed chapters count is 5, the check will determine if the chapter was completed for the first time or not by checking whether the completed chapter count has changed or remained the same.

Reply:
Done.